src/poetry/config/config.py

```diff
@@ -38,6 +43,37 @@ def int_normalizer(val: str) -> int:
     return int(val)
 
 
+def build_config_setting_validator(val: str) -> bool:
```

Comment:
Not sure if it is worth, but most of the negative paths in this function are not tested.

Reply:
I'll add some coverage.